### PR TITLE
feat: integrate chakra

### PR DESCRIPTION
**Pull request by snqb**
### Overview
This PR integrates Chakra into Quartz
https://chakra-ui.com/docs/components

Also:
- Storybook V7 with Vite
- Vite as builder

Chakra part(the new one) will be available as `import { useDisclosure } from "@logicalclock/quartz/chakra"`

image
image

Storybook starts fast:
image

### How to test:
in hopsworks repo do the `yarn link ../quartz` and in quartz either `yarn build` or `yarn start`

### What sucks:
- the final bundle size is big(it's fine btw, but looks bad)
- storybook is janky right now

### CODE REVIEW

File: src/components/navigation/context/buildComponentsTree.ts [hunk 1]
```diff
@@ -12,12 +11,6 @@ const buildComponentsTree = (
     return tree.map((node) => buildComponentsTree(node, isFirstLevel));
   }
 
-  if (isComponentNode(tree)) {
-    return React.cloneElement(tree.component!, {
-      key: tree.id,
```
Comment (snqb) [inline src/components/navigation/context/buildComponentsTree.ts:17]: it's not used anyways

File: package.json [hunk 2]
```diff
@@ -6,30 +6,28 @@
   "author": "logicalclocks",
   "license": "MIT",
   "repository": "logicalclocks/quartz",
-  "main": "dist/index.js",
-  "module": "dist/index.modern.mjs",
+  "main": "./dist/index.mjs",
```
Comment (snqb) [inline package.json:9]: we're full esm in hopsworks anyways

File: src/components/alternative-header/stories.tsx [hunk 3]
```diff
@@ -34,16 +33,16 @@ const Template: Story<AlternativeHeaderProps> = (props) => {
...
-          <Value as="span" lineHeight="13px">
-            38
-          </Value>
-        ),
-        onClick: () => setActive('last tab'),
-      },
+      // {
+      //   title: 'last tab',
+      //   isActive: active === 'last tab',
+      //   altContent: (
```
Comment (snqb) [inline src/components/alternative-header/stories.tsx:39]: I don't have a slightest clue why this thing breaks storybooks all the time, but it does

File: src/theme-chakra/ChakraThemeProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { ChakraProvider, extendTheme } from '@chakra-ui/react';
+import React, { PropsWithChildren } from 'react';
+
+const theme = extendTheme({
+  config: {
+    cssVarPrefix: 'quartz',
```
Comment (snqb) [inline src/theme-chakra/ChakraThemeProvider.tsx:6]: colors generated will be called `--quartz-colors-grayShade1` for example

File: src/components/navigation/context/buildComponentsTree.ts (hunk 1 above)
Comment (esonhey) [reply src/components/navigation/context/buildComponentsTree.ts:17]: what do you mean by it's not used?
you mean the if statement would never be `true`?

File: src/components/navigation/context/getNode.ts [hunk 5]
```diff
@@ -15,10 +14,6 @@ const getNode = (
     );
   }
 
-  if (isComponentNode(tree)) {
```
Comment (esonhey) [inline src/components/navigation/context/getNode.ts:18]: won't this be true if tree is single node?

File: src/components/navigation/context/getPathInTree.ts [hunk 6]
```diff
@@ -1,14 +1,8 @@
-import isComponentNode from '../utils';
-
 const getPathInTree = (
   tree: any,
   callback: (node: any) => boolean,
   pathKey = 'title',
 ): string[] => {
-  if (isComponentNode(tree)) {
```
Comment (esonhey) [inline src/components/navigation/context/getPathInTree.ts:8]: confused how tree would n't be a single Node as this `getPathInTree` is a recursive function you see in line 15 it's calling itself recursively

**Review by esonhey — COMMENTED** [review]
@snqb haven't run the code yet, but I got confused for changes you made for navigation, maybe there are some dead code path that I haven't figured it out yet.
If you are sure about the changes I can make deep look into it :)

**Comment by snqb** [comment]
@ehsan-github this is the piece that we used to have to render components in a navigation tree
there was one case before and it's no longer present in the codebase

you would construct a tree node with `component` instead of `children`
image

it was causing type problems and I wanted to delete it for a while
so that's why I removed this

**Comment by esonhey** [comment]
@snqb Ok I see. I found more redundant logic in Navigation component as well but let's clean it in another PR.

File: .storybook/main.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    '@storybook/addon-links',
+    '@storybook/addon-essentials',
+    '@storybook/addon-interactions',
+    'storybook-dark-mode',
+    '@chakra-ui/storybook-addon',
+  ],
+  framework: {
+    name: '@storybook/react-vite',
+    options: {},
+  },
```
Comment (esonhey) [inline .storybook/main.ts:18]: @snqb removed 
```
features: {
    emotionAlias: false,
  },
  ```
  which doesn't exist as an option for features getting types from StorybookConfig

**Review by esonhey — APPROVED** [review]

**Pull request merged**